src/Crypto/Lol/Applications/SymmSHE.hs

```diff
@@ -94,7 +94,7 @@ type EncryptCtx t m m' z zp zq =
   (Mod zp, Ring zp, Ring zq, Lift zp (ModRep zp),
    Reduce z zq, Reduce (LiftOf zp) zq,
    CElt t zq, CElt t zp, CElt t z, CElt t (LiftOf zp),
-   m `Divides` m')
+   m `Divides` m', Random zq)
```

Comment:
New constraints should be at reasonable locations, with the rest of the bare zq constraints.